data/update_script.sh.erb

```diff
@@ -51,30 +53,36 @@ echo 'PULLPREVIEW_PUBLIC_IP=<%= locals.public_ip %>' >> $PULLPREVIEW_ENV_FILE
 echo 'PULLPREVIEW_URL=<%= locals.url %>' >> $PULLPREVIEW_ENV_FILE
 echo "PULLPREVIEW_FIRST_RUN=$PULLPREVIEW_FIRST_RUN" >> $PULLPREVIEW_ENV_FILE
 
+if [ -n "${PULLPREVIEW_ENV_VARS}" ] ; then
+while read -d, -r pair; do
+  IFS='=' read -r key val <<<"$pair"
+  echo "$key=$val" >> $PULLPREVIEW_ENV_FILE
+done <<<"$PULLPREVIEW_ENV_VARS,"
+fi
+
 set -o allexport
 source $PULLPREVIEW_ENV_FILE
 set +o allexport
 
 cd /
 
-sudo rm -rf "$APP_PATH"
-sudo mkdir -p "$APP_PATH"
-sudo chown -R ec2-user.ec2-user "$APP_PATH"
-tar xzf "$1" -C "$APP_PATH"
+sudo rm -rf "/tmp$APP_PATH"
+sudo mkdir -p "$APP_PATH" "/tmp$APP_PATH"
+sudo chown -R ec2-user.ec2-user "$APP_PATH" "/tmp$APP_PATH"
+tar xzf "$1" -C "/tmp$APP_PATH"
+rsync -auz "/tmp$APP_PATH" / --remove-source-files --delete
```

Comment:
This can be pulled to upstream: Makes sure that `"$APP_PATH"` does not end up being a zombie path when docker is still running on it with mirrored volumes attached.